scripts/vaccination_alerts/utils.ts

```diff
@@ -58,3 +63,56 @@ export function getUpdatedVaccinationInfo(
   }
   return updatedInfoByFips;
 }
+
+export function readVaccinationAlerts(
+  filePath: string,
+): RegionVaccinePhaseInfoMap {
+  const data = fs.readFileSync(filePath, 'utf8');
+  return JSON.parse(data);
+}
+
+/**
+ * Given a list of [email, fips] pairs, returns a list of emails grouped
+ * by FIPS code.
+ */
+export function groupByFips(
+  emailFipsList: EmailFips[],
+): { [fipsCode: string]: string[] } {
+  return _.chain(emailFipsList)
+    .groupBy(([email, fips]) => fips)
+    .map((values, key) => [key, values.map(_.first)])
+    .fromPairs()
+    .value();
+}
+
+/**
+ * Returns the list of locations that we need to notify of vaccination information
+ * changes. For now, is just the list of keys in the `vaccination-alerts.json` file.
+ */
+export function getLocationsToAlert(
+  vaccinationAlertInfo: RegionVaccinePhaseInfoMap,
+): FipsCode[] {
+  return _.keys(vaccinationAlertInfo);
+}
+
+/**
+ * Returns the list of FIPS codes that the user should be notified given a list of
+ * locations with vaccination updates. For now, we return the list of states that
+ * contains the user subscriptions, even if the user is not subscribed to the state
+ * directly.
+ *
+ * Example: The user is subscribed to ['42001', '29005', '56', '12', '35620'], and we
+ * have updates for ['29', '01'], then we will notify the user for the changes in
+ * vaccination information for '29', even if the user is not subscribed to alerts for
+ * '29' directly.
+ */
+export function getUserLocationsToAlert(
+  userLocations: FipsCode[],
+  locationsToAlert: FipsCode[],
+): FipsCode[] {
+  return _.chain(userLocations)
+    .map(fipsCode => fipsCode.substr(0, 2))
```

Comment:
I think this will do the wrong thing for metros since metros (apparently?) have random fips codes...  New York City metro has fips 35620, so we would end up emailing them when New Mexico (fips 35) gets new vaccine info. 😬 

So I assume we need to look up the region in regions_db and check if it's a county, and then check for `region.state.fipsCode` or whatever.

Reply:
Alternatively, we could just drop this logic for now, since 88% of people that sign up for a county also sign up for the state, and that'll probably increase with the new vaccination signup UI.

Reply:
That's a good point, I tried to simplify but I think that the correct way is to fetch the state of the region using our regions API.